src/operations/write.ts

```diff
@@ -61,6 +63,56 @@ export const write = (
   return result;
 };
 
+export const writeFragment = (
+  store: Store,
+  query: DocumentNode,
+  data: Data
+) => {
+  if (process.env.NODE_ENV !== 'production' && query.definitions.length > 1) {
+    throw new Error(
+      'You can only pass one fragment when writing to a fragment.'
+    );
+  }
+
+  (query.definitions as FragmentDefinitionNode[]).forEach(
+    (fragment: FragmentDefinitionNode) => {
+      if (
+        process.env.NODE_ENV !== 'production' &&
+        fragment.kind !== Kind.FRAGMENT_DEFINITION
+      ) {
+        throw new Error(
+          `You passed a ${fragment.kind} instead of a ${Kind.FRAGMENT_DEFINITION} to writeFragment.`
+        );
+      }
+
+      const select = getSelectionSet(fragment);
+      const fieldName = getFragmentTypeName(fragment);
+      const writeData = { ...data, __typename: fieldName } as Data;
+      const fieldKey = keyOfEntity(writeData) as string;
+
+      if (process.env.NODE_ENV !== 'production' && !fieldKey) {
+        throw new Error(
+          `You have to pass an "id" or "_id" with your writeFragment data.`
+        );
+      }
+
+      const entity = store.findOrCreate(fieldKey);
+      writeSelection(
+        {
+          store,
+          variables: {},
+          fragments: {},
```

Comment:
This reminds me! Eventually we have to handle the case where more fragments are passed onto writeFragment but let's take care of that in the future in a separate PR